src/Payment.ts

```diff
@@ -3,159 +3,249 @@ import { Utils } from './Utils'
+   * @param options.maximumFees Max. transfer fees user if willing to pay.
+   */
+  public async getPath (
     network: string,
     accountA: string,
     accountB: string,
     value: number | string,
-    { decimals, maximumHops, maximumFees}: PaymentOptions = {}
-  ): Promise<any> {
-    const { utils, currencyNetwork } = this
-    const url = `networks/${network}/path-info`
-    return currencyNetwork.getDecimals(network, decimals)
-      .then(dec => {
-        const data = {
-          from: accountA,
-          to: accountB,
-          value: utils.calcRaw(value, dec)
-        }
-        if (maximumFees) {
-          data['maxFees'] = maximumFees
-        }
-        if (maximumHops) {
-          data['maxHops'] = maximumHops
-        }
-        return utils.fetchUrl(url, {
-          method: 'POST',
-          headers: new Headers({ 'Content-Type': 'application/json' }),
-          body: JSON.stringify(data)
-        })
-        .then(({ estimatedGas, fees, path }) => ({
-          estimatedGas,
-          path,
-          maxFees: utils.formatAmount(fees, dec)
-        }))
-        .catch(e => Promise.reject(e))
+    options: PaymentOptions = {}
+  ): Promise<PathObject> {
+    try {
+      const { _currencyNetwork, _utils, _user } = this
+      let { decimals, maximumHops, maximumFees} = options
+      decimals = await _currencyNetwork.getDecimals(network, decimals)
+      const data = {
+        from: accountA,
+        to: accountB,
+        value: this._utils.calcRaw(value, decimals)
+      }
+      if (maximumFees) {
+        data['maxFees'] = maximumFees
+      }
+      if (maximumHops) {
+        data['maxHops'] = maximumHops
+      }
+      const endpoint = `networks/${network}/path-info`
+      const { estimatedGas, fees, path } = await _utils.fetchUrl<PathRaw>(endpoint, {
+        method: 'POST',
+        headers: new Headers({ 'Content-Type': 'application/json' }),
+        body: JSON.stringify(data)
       })
+      return {
+        estimatedGas,
+        path,
+        maxFees: _utils.formatAmount(fees, decimals)
+      }
+    } catch (error) {
+      return Promise.reject(error)
+    }
   }
 
-  public get (networkAddress: string, filter?: object): Promise<any> {
-    const mergedFilter = Object.assign({ type: 'Transfer' }, filter)
-    return this.event.get(networkAddress, mergedFilter)
+  /**
+   * Returns transfer logs of loaded user in a specified currency network.
+   * @param network Address of currency network.
+   * @param filter Event filter object. See `EventFilterOptions` for more information.
+   */
+  public get (
+    network: string,
+    filter: EventFilterOptions = {}
+  ): Promise<TLEvent[]> {
+    return this._event.get(network, {
+      ...filter,
+      type: 'Transfer'
+    })
   }
 
-  public confirm (rawTx): Promise<string> {
-    return this.user.signTx(rawTx).then(signedTx => this.transaction.relayTx(signedTx))
+  /**
+   * Signs and relays a raw transaction as returned by `prepare`.
```

Comment:
Signs a raw transaction as returned by `prepare` and relays the signed transaction.

Reply:
done